File: Go/1.two_sum.go
package main

func twoSum(nums []int, target int) []int {
	if nums == nil {
		return []int{}
	}

	m := make(map[int]int)
	for i, v := range nums {
		if val, ok := m[target-v]; ok {
			return []int{val, i}
		}
		m[v] = i
	}
	return []int{}
}

File: Python/1.two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if not nums:
            return []
        d = {}
        for i, v in enumerate(nums):
            if target - v in d:
                return [d[target - v], i]
            d[v] = i
        return []
